manifests/apache.pp

```diff
@@ -23,20 +22,23 @@
   $apache_cert = "${pki_dir}/certs/katello-apache.crt"
   $apache_key  = "${pki_dir}/private/katello-apache.key"
   # This variable is unused but considered public API
-  $apache_ca_cert = $certs::katello_server_ca_cert
+
+  $apache_cert_path = "${certs::ssl_build_dir}/${hostname}/${apache_cert_name}"
 
   if $server_cert {
-    cert { $apache_cert_name:
-      ensure         => present,
-      hostname       => $hostname,
-      cname          => $cname,
-      generate       => $generate,
-      deploy         => false,
-      regenerate     => $regenerate,
-      custom_pubkey  => $server_cert,
-      custom_privkey => $server_key,
-      custom_req     => $server_cert_req,
-      build_dir      => $certs::ssl_build_dir,
+    file { "${apache_cert_path}.crt":
```

Comment:
This looks the wrong way around: it copies from `/etc/pki/certs/katello-apache.crt` to the build directory. But what's the point of that? Then later we have `certs::keypair` which deploys from the build directory to this location. Isn't that redundant?

Reply:
This creates `/root/ssl-build/example.com/example.com-apache.crt` using the custom certificate passed in by the user to the `server_cert` variable.

Reply:
Can we skip that step? Like enhancing the deploy step to do the copy directly.

Reply:
I contemplated that option. There are two considerations:

 1. The tar create function expects this to live in the build directory and I was trying not to break that functionality yet but rather re-factor and simplify the code.
 2. I've considered dropping this copying, and instead having the installer copy user input certificates to the build directory rather than the Puppet layer. This would attempt to solve the issue where users give certificates as input to the installer, and then by some process remove the custom certificates and things break. This would also allow backup / restore to be more self-contained I believe.

Reply:
@ekohl thoughts?

Reply:
Rebased -- @ekohl I am still waiting on your thoughts to my reply.